### ISSUE TITLE
Investigate using native truffle compiler

### ISSUE THREAD

**Issue by sohkai** [aragonone/product#46]
# Idea

We should use the native solc binaries to compile our contracts when we deploy them, as the JS-tooling stack is fairly insecure. We should make sure all tests pass after this native compilation.

## Technical requirements

Investigate if we need to switch completely to `truffle@5` or if we can stay on `truffle@4` for now (to avoid too much immediate work).

**Comment by facuspagnuolo** [comment]
I think Truffle 5 supports a native solidity version to compile (see [here](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-compile/compilerSupplier/loadingStrategies/Native.js)) and a [dockerized](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-compile/compilerSupplier/loadingStrategies/Docker.js) one. They introduced multiple alternatives actually.

**Comment by izqui** [comment]
AFAIK, the main issue with the truffle 5 upgrade (see [breaking changes](https://github.com/trufflesuite/truffle/releases/tag/v5.0.0)) is that we'd need to rewrite all the JS code (tests and scripts) to use web3.js 1.0. Unless we can avoid doing this migration (or we can automate it), this could end up being a huge undertaking for just one sprint.

Also given that apps are compiled and deployed by the CLI, we'd also need to upgrade the CLI to truffle 5.

**Comment by facuspagnuolo** [comment]
I gave it a try in the SVRP project that I've already migrated to Truffle 5 in it worked fine. It implied changing just two lines of code, after migrating all the tests ofc haha (see [here](https://github.com/aragon/svrp/pull/14)). I you guys think it's worth it I can take care of it and start migrating the test suites to Truffle 5.

Additionally, it took half time compiling the contracts with the binaries.
